tests/tests/relayer.rs

```diff
@@ -143,6 +141,79 @@ async fn relayer_can_download_logs() {
     eth_node_handle.shutdown.send(()).unwrap();
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn try_quorum_50_percent_syncing() {
+    let mut config = Config::local_node();
+    config.relayer = Some(relayer::Config::default());
+    let relayer_config = config.relayer.as_mut().expect("Expected relayer config");
+    let eth_node = MockMiddleware::default();
+
+    let eth_node = Arc::new(eth_node);
+    let eth_node_handle = spawn_eth_node(eth_node).await;
+
+    relayer_config.relayer = Some(vec![
+        format!("http://{}", eth_node_handle.address)
+            .as_str()
+            .try_into()
+            .unwrap(),
+        "http://localhost:8080".try_into().unwrap(),
+    ]);
+
+    // Given
+    let db = Database::in_memory();
+    let srv = FuelService::from_database(db.clone(), config)
+        .await
+        .unwrap();
+
+    // When
+    srv.await_relayer_synced().await.unwrap();
+
+    // Then
+    srv.send_stop_signal_and_await_shutdown().await.unwrap();
+    eth_node_handle.shutdown.send(()).unwrap();
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn quorum_33_percent_fail_syncing() {
+    let mut config = Config::local_node();
+    config.relayer = Some(relayer::Config::default());
+    let relayer_config = config.relayer.as_mut().expect("Expected relayer config");
+    let eth_node = MockMiddleware::default();
+
+    let eth_node = Arc::new(eth_node);
+    let eth_node_handle = spawn_eth_node(eth_node).await;
+
+    relayer_config.relayer = Some(vec![
+        format!("http://{}", eth_node_handle.address)
+            .as_str()
+            .try_into()
+            .unwrap(),
+        "http://localhost:8080".try_into().unwrap(),
+        "http://localhost:8081".try_into().unwrap(),
+    ]);
+
+    // Given
+    let db = Database::in_memory();
+    let srv = FuelService::from_database(db.clone(), config)
+        .await
+        .unwrap();
+    let max_timeout = std::time::Duration::from_secs(10);
+
+    // When
+    tokio::select! {
+        _ = srv.await_relayer_synced() => {
+            panic!("Should not have synced");
+        }
+        _ = tokio::time::sleep(max_timeout) => {
+            // Then
+            // We should not have synced
+        }
+    }
```

Comment:
Can we move this out so the "Then" section is on the same level as "Given" and "When"? Something like:

```suggestion
    // When
    has_synced = tokio::select! {
        _ = srv.await_relayer_synced() => {
            true
        }
        _ = tokio::time::sleep(max_timeout) => {
            false
        }
    }
    
    // Then
    assert!(!has_synced);
```

Reply:
Closing because I have re-do all the tests